fibsem/milling/patterning/plotting.py

```diff
@@ -108,29 +113,188 @@ def _draw_rectangle_pattern(
             p, pixel_size, image_shape
         )
 
-        rect = mpatches.Rectangle(
-            (px - width / 2, py - height / 2),  # bottom left corner
-            width=width,
-            height=height,
-            angle=math.degrees(p.rotation),
-            rotation_point=PROPERTIES["rotation_point"],
-            linewidth=PROPERTIES["line_width"],
-            edgecolor=colour,
-            facecolor=colour,
-            alpha=PROPERTIES["opacity"],
+        patch_collection = PatchCollection(
+            [
+                mpatches.Rectangle(
+                    (px - width / 2, py - height / 2),  # bottom left corner
+                    width=width,
+                    height=height,
+                    angle=math.degrees(p.rotation),
+                    rotation_point=PROPERTIES["rotation_point"],
+                    linewidth=PROPERTIES["line_width"],
+                    edgecolor=colour,
+                    facecolor=colour,
+                    alpha=PROPERTIES["opacity"],
+                )
+            ],
+            match_original=True,
         )
         if i == 1:
-            rect.set_label(f"{name}")
-        patches.append(rect)
+            patch_collection.set_label(f"{name}")
+        patch_collections.append(patch_collection)
 
-    return patches
+    return patch_collections
 
 
-def get_drawing_function(name: str) -> Callable:
-    
-    if name in ["Circle", "Bitmap", "Line", "SerialSection"]:
-        return None
-    return _draw_rectangle_pattern
+def _draw_bitmap_pattern(
```

Comment:
I think my preference would be to draw bitmap as image masks, rather than using patches. It would be similar to how the annulus are drawn:
https://github.com/DeMarcoLab/fibsem/blob/f5c4f74726ce20df735411a8994dba2f5f29ec62/fibsem/milling/patterning/plotting.py#L385

I think it would be more straightforward as they are already effectively image arrays.

Reply:
I see, that would be nicer but I don't see where `draw_pattern_shape` is actually used? Is this something you plan on implementing later?